src/panels/config/script/ha-script-editor.ts

```diff
@@ -419,6 +501,25 @@ export class HaScriptEditor extends KeyboardShortcutMixin(LitElement) {
     this._dirty = true;
   }
 
+  private _preprocessYaml() {
+    return this._config;
+  }
+
+  private async _copyYaml() {
+    const yaml = await import("js-yaml");
+    navigator.clipboard.writeText(yaml.safeDump(this._preprocessYaml()));
```

Comment:
Can't we get the YAML content from the `ha-yaml-editor` component?

Reply:
Well... it's a private property... though that can be changed...

`js-yaml` is most likely loaded anyway already, so it's a cheap operation.

Reply:
Add a getter to `ha-yaml-editor` (or make it public if it is already there)